src/main/java/in/zapr/druid/druidry/query/scan/DruidScanQuery.java

```diff
@@ -24,16 +24,25 @@
     private List<Interval> intervals;
     private List<DruidDimension> columns;
     private String resultFormat;
+    private Long limit;
+    private Boolean legacy;
 
     @Builder
-    private DruidScanQuery(@NonNull String dataSource, DruidFilter filter, Integer batchSize, @NonNull List<Interval> intervals, List<DruidDimension> columns, String resultFormat, Context context) {
+    private DruidScanQuery(@NonNull String dataSource, DruidFilter filter, Integer batchSize, @NonNull List<Interval> intervals, List<DruidDimension> columns, String resultFormat, Long limit, Boolean legacy, Context context) {
         this.filter = filter;
-        this.batchSize = batchSize;
         this.intervals = intervals;
         this.columns = columns;
         this.resultFormat = resultFormat;
         this.queryType = QueryType.SCAN;
         this.context = context;
         this.dataSource = dataSource;
+        this.legacy = legacy;
+        if (limit != null && limit > 0) {
```

Comment:
You should use **Preconditions** so that it would raise error, rather than not setting those fields when user send non-positive values.

And also write test case for these preconditions with expectedExceptions.

For ref : [Preconditions check example](https://github.com/zapr-oss/druidry/blob/a029c25151c01aca1d05e4b8a596c03ced8c7d17/src/main/java/in/zapr/druid/druidry/extensions/datasketches/aggregator/ThetaSketchAggregator.java#L35)

Reply:
@abhi-zapr Updated as per comment